packages/hooks/src/useIdAllocator.ts

```diff
@@ -1,53 +1,44 @@
-import { useEffect, useState } from 'react';
-
-import useIsomorphicLayoutEffect from './useIsomorphicLayoutEffect';
-
-let serverHandoffComplete = false;
-const registry: Map<string, number> = new Map();
-
-const genId = (prefix: string) => {
-  if (registry.has(prefix)) {
-    const val = registry.get(prefix);
-    const update = val ? val + 1 : 1;
-    registry.set(prefix, update);
-    return update;
-  } else {
-    registry.set(prefix, 1);
-    return 1;
-  }
-};
-
-/**
- * Generates an SSR-compatible unique id based on a prefix string and an optional idProp parameter.
- * @param options {prefix: string; id: string;}
- * @param options.prefix string that prefixes the generated id
- * @param options.id string that represents an already-created id
- */
-function useIdAllocator({
-  prefix,
-  id: idProp,
-}: {
+// Currently using Material UI useId hook until we can upgrade to React 18's useId
+// https://github.com/mui/material-ui/blob/master/packages/mui-utils/src/useId.ts
+import React, { useEffect, useState } from 'react';
+
+interface Params {
   prefix?: string;
   id?: string;
-}) {
-  const initialId = prefix && (serverHandoffComplete ? genId(prefix) : null);
+}
 
-  const [id, setId] = useState(initialId);
+let globalId = 0;
 
-  useIsomorphicLayoutEffect(() => {
-    if (prefix && id === null) {
-      setId(genId(prefix));
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+function useGlobalId({ id: idOverride, prefix }: Params): string {
+  const [defaultId, setDefaultId] = useState<string | number | undefined>(
+    idOverride,
+  );
 
   useEffect(() => {
-    if (serverHandoffComplete === false) {
-      serverHandoffComplete = true;
+    if (defaultId == null) {
+      // Fallback to this default id when possible.
+      // Use the incrementing value for client-side rendering only.
+      // We can't use it server-side.
```

Comment:
We just spoke about this, but I'm confused about this comment. If we're not incrementing the value server-side, then what is returned server-side? it looks like `undefined` if there is no `idOverride`?